### ISSUE TITLE
Basis: content edit/create form not full-width (main region uses the col-md-9 class)

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#4833]
**Steps To Reproduce**

1. Go to `/admin/appearance`
2. Scroll to the bottom of the page, and disable the default, OOTB "Use the administration theme when editing or creating content" setting
3. Save.
4. Add/edit any type of content...

**Actual behavior**

![basis-edit-content](https://user-images.githubusercontent.com/2423362/103144549-ef61cb80-4733-11eb-81be-662434b1290c.png)

**Expected behavior**

The main region/area when editing content should be full-width:

![basis-edit-content-full-width](https://user-images.githubusercontent.com/2423362/103144599-fccb8580-4734-11eb-8ad3-9e8578aa4916.png)

**Additional information**

- I understand that toggling the "Use the administration theme when editing or creating content" setting may make it so that the "Default Administrative Layout" layout is not used any more (should that happen though?)
- I also understand that editing content would most likely now default to the "Default Layout" layout, which has a "Sidebar" region in the same place where this "gap" appears, but I'd expect that if no blocks are being rendered in that region, then that region would not be rendered either. Right?
- Not sure if this was always the case, since I never toggle the "Use the administration theme when editing or creating content" setting on my locals when testing. Did we regress at some point? 🤔

**Comment by docwilmot** [comment]
Basis simply cannot be used as an admin theme.

**Comment by klonos** [comment]
Well, Basis is what most people will get when editing content or filing forms, if they don't have the "View the administration theme" permission 🤷 ...since it is a core theme, we should make sure that it is presentable/usable, and works as expected.

**Comment by indigoxela** [comment]
> I understand that toggling the "Use the administration theme when editing or creating content" setting may make it so that the "Default Administrative Layout" layout is not used any more (should that happen though?)

@klonos What you see is IMO correct behavior. The layout that will be used for display is also used for editing. That's WYSIWYG and that's why I prefer to edit content in the frontend theme. The form has the same width as the display will have.

The sidebar in Moscone flipped (default) isn't dynamic, it's also there when it's empty. Also, not a bug IMO. If people want a different layout, for instanced they don't need a sidebar at all, they can switch to another layout any time.

> Basis simply cannot be used as an admin theme.

@docwilmot While I largely agree on that, it's not the issue here. It's about the layout in use for content editing.

**Comment by olafgrabienski** [comment]
> Basis simply cannot be used as an admin theme.

There is a related meta issue: [Make sure Basis looks and works decently when used as an admin theme](https://github.com/backdrop/backdrop-issues/issues/2653)

**Comment by docwilmot** [comment]
> @docwilmot While I largely agree on that, it's not the issue here. It's about the layout in use for content editing.

Wrong post title then.

**Comment by klonos** [comment]
>The sidebar in Moscone flipped (default) isn't dynamic, it's also there when it's empty.

You're absolutely tight:

![image](https://user-images.githubusercontent.com/2423362/103164359-ed226e80-4812-11eb-855c-7586bb44e312.png)


🤔 ...I thought that if no blocks were to be rendered in the sidebar, then the entire region would not be rendered, allowing the main region to span the full width of the screen. For example, think a login block added to the sidebar; once you log into the site, the login block is not rendered, so the sidebar is "wasted space".

It seems that @docwilmot asked the same thing back in 2015 in #596, and got the same "works as designed" answer; I just had it wrong in my head all the time 🤷 

Anyway, you are right that this is how things work though, and we already have issues like #4025 for this, so I'm thinking to close this one as WAD, unless we want to document this for others.

**Comment by docwilmot** [comment]
So to be clear, this issue is about the layout being funny, not the theme.

But, to be clear, if you uncheck that box, the default theme **and** default layout (not admin default) then used for node edit, right? That isn't quite clear; do we need help text on that checkbox or is it a reasonable assumption?

**Comment by klonos** [comment]
>So to be clear, this issue is about the layout being funny, not the theme.

Yes. Initially, I assumed that there is something wrong with Basis; but the issue is actually with how layouts work (#596), and the same would happen if Seven was used as the frontend theme. This is the bit that needs to be documented:

- Regions without any blocks within them will still be rendered anyway (but will be "blank").
- Even if a region within a layout does have blocks, but these are hidden because of visibility conditions, the region will be rendered anyway (without any block in it).

> is it a reasonable assumption?

I would argue that it is not, but we've established that this is how things work. The reasonable thing that happens (in the context of what the checkbox does, or its consequences if you like) is this:

- Toggling the "Use the administration theme when editing or creating content" setting may make it so that the "Default Administrative Layout" layout is not used any more.
- Editing content would _most likely_ now default to the "Default Layout" layout.

**Comment by klonos** [comment]
...and to be clear, the 2 first bullet points in my previous comment do not belong in any way as help/indication in the theme switching admin form. If we did want to make it clear in the UI, then we could do something like this in the "Manage blocks" form for regions w/o any blocks:

![screencapture-demos3-softaculous-Backdrop-CMSmdy5ujkvb2-admin-structure-layouts-manage-default-2020-12-28-01_01_42](https://user-images.githubusercontent.com/2423362/103181295-5e603100-48a8-11eb-9c82-f59ab6d24984.png)

**Comment by indigoxela** [comment]
@klonos I don't think that we can add such a help text to sidebar regions automatically (based on which information?). There's at least one contrib layout that *does* collapse an empty sidebar: https://backdropcms.org/project/two_column_flexible

And custom layouts might handle things dynamically, too. So we can't assume that a text like "this region will always be rendered..." would always be correct.

**Comment by klonos** [comment]
> So we can't assume...

Fair enough 👍 ...I'm making this a (theming/layout) documentation task then.
